src/audio/kpb.c

```diff
@@ -520,6 +523,11 @@ static void kpb_init_draining(struct comp_data *kpb, struct kpb_client *cli)
 {
 	uint8_t is_sink_ready = (kpb->clients[cli->id].sink->sink->state
 				 == COMP_STATE_ACTIVE) ? 1 : 0;
+	size_t history_depth = cli->history_depth;
+	struct hb *buff = &kpb->history_buffer;
+	size_t buffered = 0;
+	size_t read_offset;
+	struct hb *first_buff = buff;
```

Comment:
@mrajwa no need for this you can always use &cd->history_buffer to point to the first buffer